### PR TITLE
added more mysql queries to suit the need for peekTasks()

### PR DESCRIPTION
**Pull request by ywang282**
/tasks path assumes either both STATE and CLUSTER_NAME or either one of them is specified in the parameter. However, peekTasks() fails to get any matching rows in the db when only one of the field is present. Because the default value for both STATE and CLUSTER_NAME is NULL and mysql is matching for CLUSTER_NAME = name AND STATE = null (eg, when STATE is not provided).

### CODE REVIEW

**Comment by shuzhang1989** [comment]
do you also want to change the http layer to not throw exception?

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java [hunk 1]
```diff
@@ -369,9 +369,26 @@ public boolean finishTaskAndEnqueuePendingTask(final long id,
 
   @Override
   public List<Task> peekTasks(final String clusterName, final Integer state) {
-    Query query = getEntityManager()
-        .createNamedQuery("task.peekTasks")
+    Query query;
+    if (clusterName != null && state != null){
```
Comment (shuzhang1989) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java]: nit: space between ) and {

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java [hunk 2]
```diff
@@ -369,9 +369,26 @@ public boolean finishTaskAndEnqueuePendingTask(final long id,
...
+      query = getEntityManager()
+        .createNamedQuery("task.peekTasksFromClusterWithState")
         .setParameter("state", state).setParameter("name", clusterName);
+    }
+    else if (clusterName == null){
```
Comment (shuzhang1989) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java]: nit: put it in the last line
if {
} else if {
}

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java (hunk 1 above)
Comment (ywang282) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java]: fixed

File: controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java (hunk 2 above)
Comment (ywang282) [inline controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java]: fixed

**Comment by shuzhang1989** [comment]
awesome!

**Pull request merged**